### PR TITLE
RATEST:224:E2E Test: Running reports.

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/RATEST-224

### CODE REVIEW

**Comment by sherrif10** [comment]
cc @kdaud Looking forward for approval, otherwise we can request reviews from @dkayiwa

**Comment by kdaud** [comment]
are you sure you addressed the comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/189#discussion_r739095292)?

**Comment by sherrif10** [comment]
I used setTextToFieldNoEntry()  method which does the same thing.

**Comment by kdaud** [comment]
do we really need the waiter method when setting dates?

**Comment by sherrif10** [comment]
Sure like  I said above in my previous comments, this waiter is helping us
to retrieve that element without throwing in reachable clickable elements.

**Comment by kdaud** [comment]
are the dates filed take long to long on the page?

**Comment by sherrif10** [comment]
Sure the dates take long to be retrieved hence throwing elements not
clickable.

File: src/test/java/org/openmrs/contrib/qaframework/automation/ReportSteps.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+public class ReportSteps extends Steps {
+
+	private SystemAdministrationPage systemAdministrationPage;
+	private AdministrationPage administrationPage;
+	private ManageReportsPage manageReportsPage;
+	private ReportHistoryPage reportHistoryPage;
+	private RunReportPage runReportsPage;
+	private RenderDefaultReportPage renderDefaultReportPage;
+	private static final String startDate = "05 / 07 / 2008";
+	private static final String endDate = "20 / 06 / 2020";
```
Comment (kdaud) [inline src/test/java/org/openmrs/contrib/qaframework/automation/ReportSteps.java:27]: did you just forget to align the static final variable names to right case letter?

File: src/test/java/org/openmrs/contrib/qaframework/automation/ReportSteps.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+public class ReportSteps extends Steps {
+
+	private SystemAdministrationPage systemAdministrationPage;
+	private AdministrationPage administrationPage;
+	private ManageReportsPage manageReportsPage;
+	private ReportHistoryPage reportHistoryPage;
+	private RunReportPage runReportsPage;
+	private RenderDefaultReportPage renderDefaultReportPage;
+	private static final String STARTDATE = "05 / 07 / 2008";
```
Comment (kdaud) [inline src/test/java/org/openmrs/contrib/qaframework/automation/ReportSteps.java:26]: isn't this supposed to be `START_DATE`?

File: src/test/java/org/openmrs/contrib/qaframework/automation/ReportSteps.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	private static final String ENDDATE = "20 / 06 / 2020";
```
Comment (kdaud) [inline src/test/java/org/openmrs/contrib/qaframework/automation/ReportSteps.java:27]: isn't this supposed to be `END_DATE`?

File: src/test/resources/features/refapp-2.x/015-reports/reports.feature [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+     
+  @selenium
+  @report
+  Scenario: Report Management
+    When user clicks on advanced administration page
+    And user clicks on report administration link
+    Then the system loads manage reports page
+    And user clicks on run button from manage Reports page
+    And user enters start date
+    Then user enters end date
```
Comment (kdaud) [inline src/test/resources/features/refapp-2.x/015-reports/reports.feature:15]: isn't this gherkin syntax supposed to be `And use enter end date`?

Comment (sherrif10) [reply src/test/resources/features/refapp-2.x/015-reports/reports.feature:15]: Nope thanks that scenario isnt well readable, it should be **user** not **use**

Comment (kdaud) [reply src/test/resources/features/refapp-2.x/015-reports/reports.feature:15]: I meant the gherkin syntax of `Then` Vs `And`

File: src/test/java/org/openmrs/contrib/qaframework/automation/ReportSteps.java (hunk 3 above)
Comment (sherrif10) [reply src/test/java/org/openmrs/contrib/qaframework/automation/ReportSteps.java]: Resolved thanks.

**Comment by sherrif10** [comment]
@kdaud  Lets finish up this, Hopefully have addressed all your comments.

**Review by kdaud — APPROVED** [review]
Thanks @sherrif10 for the great work.

 As @dkayiwa takes final reviews on the PR, its high time you finished [RATEST-258](https://issues.openmrs.org/browse/RATEST-258) because we really need selenium legacy tests migrate to qaframework.

**Comment by kdaud** [comment]
Hey @dkayiwa we have this beautiful daughter who eagerly desires to be married to master and waiting for you to perform the action.

**Pull request merged**